app/models/custom_filter.rb

```diff
@@ -24,6 +24,28 @@ class CustomFilter < ApplicationRecord
   enum filter_type: { conversation: 0, contact: 1, report: 2 }
   validate :validate_number_of_filters
 
+  def records_count
+    fetch_record_count_from_redis || set_record_count_in_redis
+  end
+
+  def filter_records
```

Comment:
Should we use the filter.user_id here?
also lets pass filter.account as an argument that relying on current account in the filter service

let's try our best to keep usage of Current into controllers alone.

Reply:
`Current.account` usage has been removed.

Reply:
Any reason we need the user in the first place?

Reply:
The main filter query involves the user in `FilterService.` for assigned conversations. We can update the query, but it affects automation as well as custom filters, so we kept it for another ticket.